testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksDelete.java

```diff
@@ -79,28 +84,263 @@ public void testDeletionVectors()
-                    .hasMessageMatching(".* Table .* does not exist");
-            assertQueryFailure(() -> onTrino().executeQuery("SHOW COLUMNS FROM delta.default." + tableName))
-                    .hasMessageMatching(".* Table .* does not exist");
-            assertQueryFailure(() -> onTrino().executeQuery("DESCRIBE delta.default." + tableName))
-                    .hasMessageMatching(".* Table .* does not exist");
+            assertThat(onTrino().executeQuery("SELECT column_name FROM delta.information_schema.columns WHERE table_schema = 'default' AND table_name = '" + tableName + "'"))
+                    .contains(row("a"), row("b"));
+            assertThat(onTrino().executeQuery("SELECT version, operation FROM delta.default.\"" + tableName + "$history\""))
+                    .contains(row(0, "CREATE TABLE"), row(1, "WRITE"), row(2, "DELETE"));
+            assertThat(onTrino().executeQuery("SHOW COLUMNS FROM delta.default." + tableName))
+                    .contains(row("a", "integer", "", ""), row("b", "integer", "", ""));
+            assertThat(onTrino().executeQuery("DESCRIBE delta.default." + tableName))
+                    .contains(row("a", "integer", "", ""), row("b", "integer", "", ""));
+
+            // TODO https://github.com/trinodb/trino/issues/17063 Use Delta Deletion Vectors for row-level deletes
             assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (3, 33)"))
-                    .hasMessageMatching(".* Table .* does not exist");
+                    .hasMessageMatching(".* Table .* requires Delta Lake writer version 7 which is not supported");
             assertQueryFailure(() -> onTrino().executeQuery("DELETE FROM delta.default." + tableName))
-                    .hasMessageMatching(".* Table .* does not exist");
+                    .hasMessageMatching(".* Table .* requires Delta Lake writer version 7 which is not supported");
             assertQueryFailure(() -> onTrino().executeQuery("UPDATE delta.default." + tableName + " SET a = 3"))
-                    .hasMessageMatching(".* Table .* does not exist");
+                    .hasMessageMatching(".* Table .* requires Delta Lake writer version 7 which is not supported");
             assertQueryFailure(() -> onTrino().executeQuery("MERGE INTO delta.default." + tableName + " t USING delta.default." + tableName + " s " +
                     "ON (t.a = s.a) WHEN MATCHED THEN UPDATE SET b = -1"))
-                    .hasMessageMatching(".* Table .* does not exist");
+                    .hasMessageMatching(".* Table .* requires Delta Lake writer version 7 which is not supported");
+        }
+        finally {
+            dropDeltaTableWithRetry("default." + tableName);
+        }
+    }
+
+    // TODO: Add DELTA_LAKE_OSS group once they support creating a table with deletion vectors
```

Comment:
Upcoming OSS Delta Lake 2.4.0 supports writing deletion vectors.

Reply:
can you put "2.4.0" in the TODO comment?
perhaps

```
// TODO: Upgrade Delta OSS to 2.4.0 to get support for deletion vectors and add DELTA_LAKE_OSS group

Reply:
Sorry, my above comment is obsolete. The current testing version is 2.4.0. 

It seems the internal implementation of deletion vectors between Databricks and OSS Delta Lake differ. Some tests fail on OSS Delta Lake. I'm investigating the cause.

Reply:
Does this mean we can read Databricks deletion vectors, but fail to read OSS Delta's?

Reply:
There were query failure and silent correctness issue. The query failure happened when checkpoint parquet files didn't contain deletion vector. The silent correctness issue happened due to `TransactionLogAccess#activeAddEntries`. Deletion vectors store both `add` and `remove` entries in a file. 
* The order of Databricks is `remove` → `add`. The active `add` entry was kept and returned the expected result. 
* The order of OSS Delta Lake is `add` → `remove`. The active `add` entry was removed and returned incorrect result.

Changed to avoid depending on the order.